src/VecSim/algorithms/hnsw/hnsw_batch_iterator.cpp

```diff
@@ -186,7 +186,7 @@ VecSimQueryResult_List HNSW_BatchIterator::getNextResults(size_t n_res,
     batch = this->prepareResults(top_candidates, n_res);
 
     this->updateResultsCount(VecSimQueryResult_Len(batch));
-    if (this->getResultsCount() == this->index_wrapper->indexSize()) {
+    if (this->getResultsCount() == this->index_wrapper->indexLabelCount()) {
```

Comment:
do we need to verify this one on `multi` scenario?

Reply:
What do you mean by verify?